src/components/timeline/index.js

```diff
@@ -52,21 +59,97 @@ type StateProps = {|
   +panelLayoutGeneration: number,
   +zeroAt: Milliseconds,
   +timelineType: TimelineType,
+  +hiddenTrackCount: HiddenTrackCount,
 |};
 
 type DispatchProps = {|
   +changeGlobalTrackOrder: typeof changeGlobalTrackOrder,
   +commitRange: typeof commitRange,
   +updatePreviewSelection: typeof updatePreviewSelection,
   +changeTimelineType: typeof changeTimelineType,
+  +changeRightClickedTrack: typeof changeRightClickedTrack,
 |};
 
 type Props = ConnectedProps<OwnProps, StateProps, DispatchProps>;
 
-class Timeline extends PureComponent<Props> {
+class TimelineSettingsGraphType extends React.PureComponent<{|
+  +timelineType: TimelineType,
+  +changeTimelineType: typeof changeTimelineType,
+|}> {
   _changeToCategories = () => this.props.changeTimelineType('category');
   _changeToStacks = () => this.props.changeTimelineType('stack');
 
+  render() {
+    const { timelineType } = this.props;
+
+    return (
+      <form>
+        <div className="timelineSettingsToggle">
+          Graph type:{' '}
+          <label className="timelineSettingsToggleLabel">
+            <input
+              type="radio"
+              name="timelineSettingsToggle"
+              className="timelineSettingsToggleInput"
+              checked={timelineType === 'category'}
+              onChange={this._changeToCategories}
+            />
+            Categories
+          </label>
+          <label className="timelineSettingsToggleLabel">
+            <input
+              type="radio"
+              name="timelineSettingsToggle"
+              className="timelineSettingsToggleInput"
+              checked={timelineType === 'stack'}
+              onChange={this._changeToStacks}
+            />
+            Stack height
+          </label>
+        </div>
+      </form>
+    );
+  }
+}
+
+class TimelineSettingsHiddenTracks extends React.PureComponent<{|
+  +hiddenTrackCount: HiddenTrackCount,
+  +changeRightClickedTrack: typeof changeRightClickedTrack,
+|}> {
+  _showMenu = (event: SyntheticMouseEvent<HTMLElement>) => {
+    const rect = event.currentTarget.getBoundingClientRect();
+    changeRightClickedTrack(null);
+    showMenu({
+      data: null,
+      id: 'TimelineTrackContextMenu',
+      position: { x: rect.left, y: rect.bottom },
+      target: event.target,
+    });
+  };
+
+  render() {
+    const { hiddenTrackCount } = this.props;
+
+    return (
+      <button
+        type="button"
+        onClick={this._showMenu}
```

Comment:
Optional: This feels a bit weird that it's not a toggle button.

Reply:
Optional because these are likely issues in the context menu library: 

STR1:
1. display the menu using the keyboard
2. without moving the cursor, press "enter" again
=> the first element is checked/unchecked, despite that it's not selected

STR2:
1. display the menu using the keyboard
2. move the cursor on a line that's not the first one (that makes the issue more visible)
3. press enter
=> notice how it looks like that the menu is "reloaded", especially we lose the selection position.

If it's not easy to fix them now, can you please file issues about these problems?

Reply:
Yeah, these are all issues with the context menu library. I filed: https://github.com/firefox-devtools/profiler/issues/1826